pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -597,6 +608,233 @@ def test_optimize(self):
         # check that sync was optimized
         self.assertTrue(optimized)
 
+    def test_sync_advisory_new_version(self):
```

Comment:
Please add tests which performs sync in other order and which checks that the advisory is preserved.